src/components/Carousel/Carousel.tsx

```diff
@@ -135,20 +143,57 @@ const Carousel: React.FC<CarouselProps> = ({
     setCurIndex(length - 1)
   }, [length])
 
+  /**
+   * Pause
+   */
+  const pause = useRef(false) // Flag to pause carousel loop
+
   /**
    * Slide to the previous carousel item
    */
   const slidePrev = useCallback(() => {
+    if (pause.current) return
     setCurIndex((curIndex: number) => getPrevIndex(length, curIndex))
   }, [length])
 
   /**
    * Slide to the next carousel item
    */
   const slideNext = useCallback(() => {
+    if (pause.current) return
     setCurIndex((curIndex: number) => getNextIndex(length, curIndex))
   }, [length])
 
+  const selectSlide = useCallback(
+    (index: number) => {
```

Comment:
In my opinion, it would be beneficial if 'selectable' and 'onClickSelectedItem' are independent.
- 'Selectable' could handle whether the curIndex can be changed through a click
- 'onClickSelectedItem' is the event when the currently selected (frontmost) carousel item is clicked
```javascript
(index: number) => {
    if (selectable) setCurIndex(index)
    if (onClickSelectedItem && curIndex === index) onClickFocusedItem(index)
  }
```

Reply:
I agree to make independent the event and the selectable property. Done